torch/csrc/autograd/engine.cpp

```diff
@@ -443,7 +432,7 @@ void Engine::reentrant_thread_init() {
     // set the local_ready_queue to the ready queue on the graph_task->owner_ device
     local_ready_queue = ready_queue_by_index(graph_task->cpu_ready_queue_, graph_task->owner_);
     total_depth = graph_task->reentrant_depth_;
-    thread_main(graph_task, /* reentrant thread*/ true);
+    thread_main(graph_task, /* device_thread */ false);
```

Comment:
It's a bit confused on the argument changed here, device threads can also have reentrant backwards right?

Reply:
I would agree here. Since we have multiple use case for this, it might be clearer to just describe what it does.
Or just remove this flag that is completely redundant with the first arguments :D